### PR TITLE
fix(core,kit): ensure module transpilation paths are dirs

### PR DESCRIPTION
**Pull request by harlan-zw**
### 🔗 Linked issue

- https://stackblitz.com/edit/nuxt-starter-nb7fac?file=nuxt.config.ts
- https://stackblitz.com/edit/nuxt-starter-f7ehnh?file=nuxt.config.ts

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

When Nuxt installs modules it adds the modules entire directory to be transpiled. 

When using an absolute path to a module such as `/home/user/my-project/node_modules/my-module/dist/module.mjs` though, this transpiling won't take effect, as we're only targeting the module file itself. (see stackblitz demos)

To get around this we check if the path is to a file, if so we use the directory it's in. 

This is fairly edge-case as you can for most modules you would just use the module name itself. ]

Normally I would just say that this should be fixed in userland, but in this instance, it creates quite confusing bugs that aren't easy to debug, so I think it's worth fixing in Nuxt.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/kit/src/module/install.ts [hunk 1]
```diff
@@ -22,7 +23,7 @@ export async function installModule (moduleToInstall: string | NuxtModule, _inli
   }
 
   if (typeof moduleToInstall === 'string') {
-    nuxt.options.build.transpile.push(moduleToInstall)
+    nuxt.options.build.transpile.push(extname(moduleToInstall) ? dirname(moduleToInstall) : moduleToInstall)
```
[the remark this answers is not in the document]
Comment (harlan-zw) [reply packages/kit/src/module/install.ts]: Much better idea, thanks. Is it safe to use `node:fs` within the `installModule` @nuxt/kit util :thinking:

[the remark this answers is not in the document]
Comment (danielroe) [reply packages/kit/src/module/install.ts]: yes - we can rely on this being run in node 👍

File: packages/kit/src/module/install.ts [hunk 2]
```diff
@@ -35,6 +37,17 @@ export async function installModule (moduleToInstall: string | NuxtModule, _inli
 
 // --- Internal ---
 
+export const normaliseModuleTranspilePath = (p: string) => {
```
Comment (danielroe) [inline packages/kit/src/module/install.ts:40]: probably we want this to be US English. 🥲

```suggestion
export const normalizeModuleTranspilePath = (p: string) => {
```

Comment (harlan-zw) [reply packages/kit/src/module/install.ts]: haha yup, always catches me out

Comment (danielroe) [reply packages/kit/src/module/install.ts]: (it'll have to be the other files too)

**Review by danielroe — APPROVED** [review]

**Pull request merged**